### PR TITLE
Terraform code trn generations

### PR DESCRIPTION
**Pull request by enformatique11**
Produced the terraform code to test the trn generation API  deployment locally.
It has been successful - link as follows: https://portal.azure.com/#@platform.education.gov.uk/resource/subscriptions/9816b7b0-58ca-4972-b25b-c7dbd3ee1c6d/resourceGroups/s165d01-trngen-dv-rg/overview

### CODE REVIEW

**Comment by enformatique11** [comment]
Tested the code locally and it deploys the required resources.

File: terraform/app.tf [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ]
+  }
+}
+
+# Postgress Server
+resource "azurerm_postgresql_flexible_server" "postgres-server" {
+  name                   = local.postgres_server_name
+  location               = data.azurerm_resource_group.resource_group.location
+  resource_group_name    = data.azurerm_resource_group.resource_group.name
+  version                = "12"
```
Comment (saliceti) [inline terraform/app.tf:19]: Please update to latest: 14

File: Makefile [hunk 2]
```diff
@@ -56,6 +56,20 @@ validate-keyvault-secret: read-keyvault-config install-fetch-config set-azure-ac
 	bin/fetch_config.rb -s azure-key-vault-secret:${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} -d quiet \
 		&& echo Data in ${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} looks valid
 
+terraform-init:
+	$(if $(IMAGE_TAG), , $(eval export IMAGE_TAG=main))
+	[[ "${SP_AUTH}" != "true" ]] && az account set -s $(AZURE_SUBSCRIPTION) || true
+	terraform -chdir=terraform init -backend-config workspace_variables/${DEPLOY_ENV}.backend.tfvars $(backend_config) -upgrade -reconfigure
+
+terraform-plan: terraform-init# make dev terraform-plan
```
Comment (saliceti) [inline Makefile:64]: Please add a whitespace before the # signs

File: terraform/app.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  os_type             = "Linux"
+  sku_name            = "B1"
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+# Postgress Server
```
Comment (saliceti) [inline terraform/app.tf:14]: This comment is not very useful

File: terraform/data.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  resource_group_name = data.azurerm_resource_group.resource_group.name
+}
+
+data "azurerm_key_vault_secrets" "secrets" {
+  key_vault_id = data.azurerm_key_vault.vault.id
+}
+
+data "azurerm_key_vault_secret" "secrets" {
+  key_vault_id = data.azurerm_key_vault.vault.id
+  for_each     = toset(data.azurerm_key_vault_secrets.secrets.names)
```
Comment (saliceti) [inline terraform/data.tf:16]: As convention,`for_each` should me moved to first line

File: terraform/output.tf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+output "postgres_server_name" {
```
Comment (saliceti) [inline terraform/output.tf:1]: Do we need these outputs? If not they can be removed

File: terraform/app.tf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  }
+}
+
+# database
+resource "azurerm_postgresql_flexible_server_database" "postgres-database" {
+  name      = local.postgres_database_name
+  server_id = azurerm_postgresql_flexible_server.postgres-server.id
+}
+
+resource "azurerm_linux_web_app" "auth-server-app" {
```
Comment (saliceti) [inline terraform/app.tf:48]: Please provide the app secrets as environment variables

File: terraform/app.tf (hunk 1 above)
Comment (enformatique11) [reply terraform/app.tf:19]: done

File: Makefile (hunk 2 above)
Comment (enformatique11) [reply Makefile:64]: fixed

File: terraform/data.tf (hunk 4 above)
Comment (enformatique11) [reply terraform/data.tf:16]: done

File: terraform/output.tf (hunk 5 above)
Comment (enformatique11) [reply terraform/output.tf:1]: fixed - removed output "postgres_server_name"

File: terraform/app.tf (hunk 6 above)
Comment (enformatique11) [reply terraform/app.tf:48]: added

**Review by saliceti — APPROVED** [review]

**Pull request merged**